### PR TITLE
[KYUUBI #4218] Using DB and table name when checking Delta table schema.

### LINKED ISSUE #4218: [Bug] Delta Lake columns only show in DBeaver/Power BI/etc. in the "default" database
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

When trying to list columns in a Delta Lake table using a tool like DBeaver or Power BI the operation only works if the table is in the "default" database. If the table is in any other database, the operation will fail with a "table or view not found" error.

### Affects Version(s)

master/1.6.1

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by nousot-cloud-guy**
### _Why are the changes needed?_

To close #4218 .
This change ensures BI tools can list columns on Delta Lake tables in all schemas.

image

image

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala [hunk 1]
```diff
@@ -141,7 +141,7 @@ class CatalogShim_v2_4 extends SparkCatalogShim {
       catalog.getTablesByName(identifiers).flatMap { t =>
         val tableSchema =
           if (t.provider.getOrElse("").equalsIgnoreCase("delta")) {
-            spark.table(t.identifier.table).schema
+            spark.table(f"${db}.${t.identifier.table}").schema
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala:144]: can we use `t.identifier.toString` here?

**Comment by pan3793** [comment]
Thanks for investigating and making this PR, could you please verify the case when the db or table name contains dots?

**Comment by pan3793** [comment]
After second thought, I think we can totally remove the current workaround, and just invert the pattern match' branch order in `CatalogShim_v3_0#getColumnsByCatalog`

```patch
- catalog match {
-   case builtin if builtin.name() == SESSION_CATALOG =>
-     super.getColumnsByCatalog(...)
-   case tc: TableCatalog =>
-     ...
- }

+ catalog match {
+   case tc: TableCatalog =>
+     ...
+   case builtin if builtin.name() == SESSION_CATALOG =>
+     super.getColumnsByCatalog(...)
+ }
```

Would you please try this way? @nousot-cloud-guy

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala (hunk 1 above)
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/shim/CatalogShim_v2_4.scala:144]:
```suggestion
            spark.table(t.identifier.toString).schema
```

**Comment by nousot-cloud-guy** [comment]
> 

That seems to be working. I've updated the PR with the reverts and new commit.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master/1.6, thanks @nousot-cloud-guy

**Pull request closed without merging**